packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-listener.ts

```diff
@@ -143,8 +147,20 @@ export class ApplicationListener extends BaseListener implements IApplicationLis
   /**
    * Add one or more certificates to this listener.
    */
-  public addCertificateArns(_id: string, arns: string[]): void {
-    this.certificateArns.push(...arns);
+  public addCertificateArns(id: string, arns: string[]): void {
+    const [first, ...rest] = arns;
+    const additionalCertArns = this.certificateArns.length ? arns : rest;
+
+    if (!this.certificateArns.length && typeof first === "string") {
```

Comment:
Is this to deal with the case where an empty list was passed? I'd rather you test for that explicitly than rely on the (to me somewhat obscure) behavior that:

```ts
const [x, ...xs] = [];
// x === undefined
```

I would have expected that to throw. Obviously not, because JavaScript :), but the `typeof` check confused me before I tried this out in the REPL.

I'd much rather see a

```ts
if (arns.length === 0) { return; }
```

At the top.

Reply:
This was to defend against an empty array being passed, I agree that explicitly testing that condition is better.